sdk/python/kfp/v2/dsl/type_utils.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Utilities for component I/O type mapping."""
+
+from kfp.v2.proto import pipeline_spec_pb2
+
+# ComponentSpec I/O types to (IR) PipelineTaskSpec I/O types mapping.
+# The keys are normalized (lowercased). These are types viewed as Artifacts.
+# The values are the corresponding IR artifact types.
+_ARTIFACT_TYPES_MAPPING = {
+    # TODO: support more artifact types
+    'gcspath': 'mlpipeline.Artifact',
+    'model': 'mlpipeline.Model',
+}
+
+# ComponentSpec I/O types to (IR) PipelineTaskSpec I/O types mapping.
+# The keys are normalized (lowercased). These are types viewed as Parameters.
+# The values are the corresponding IR parameter primitive types.
+_PARAMETER_TYPES_MAPPING = {
+    'integer': pipeline_spec_pb2.PrimitiveType.INT,
+    'int': pipeline_spec_pb2.PrimitiveType.INT,
+    'double': pipeline_spec_pb2.PrimitiveType.DOUBLE,
+    'float': pipeline_spec_pb2.PrimitiveType.DOUBLE,
+    'string': pipeline_spec_pb2.PrimitiveType.STRING,
```

Comment:
Interesting... For my education, why do we need both `string` and `str`?

Reply:
Similar question for `integer` and `int`

Reply:
My 2c are as follows:

No matter what we're going to restrict the names of input/output type to a limited set of names defined in the SDK (unless the user customize their own type), which is inevitably much more restrictive than arbitrary string. Aliasing can help users bypass some errors but there're two potentially implicit cost here:

1) bifurcating the code path will make the code less manageable and harder to reason;
2) and the criterion on when to do this or not is not super clear. For example, why 'str' is an alias but 'Text' is not, in addition to other languages.

For this specific problem, I think a compile time error which explicitly states that the only type of supported parameter right now is int/string/double, and users will immediately know how to make their yaml spec work properly. WDYT?

Reply:
Thanks, I'm convinced. Changed to use a more strict list.